tools/Makefile

```diff
@@ -42,8 +42,10 @@ update-version:
 	## When manually updating major version
 	./releaser/scripts/update-version.sh
 
-.PHONY: generate_mocks
-generate_mocks:	
-	go install github.com/vektra/mockery/v2@v2.42.1
+.PHONY: generate_mocks_linux
+generate_mocks_linux:	
 	rm -Rf ../mockadmin
-	mockery
+	curl -L -o mockery.tar.gz https://github.com/vektra/mockery/releases/download/v2.42.1/mockery_2.42.1_Linux_x86_64.tar.gz
```

Comment:
There are couple antipatterns here:
1. We do not cache installation (mocking locally fails 
2. We do not clean up files
3. We do use inefficient and undocumented way of installation
4. We do not check if tool exist (making redundant installs without cache)

We tend to use `.bin` file in the repo so developers do not get tar.gz file as change when running this command.

See also ASDF docs: https://github.com/asdf-community/asdf-golang?tab=readme-ov-file#default-go-get-packages

Official docs:
https://vektra.github.io/mockery/latest/installation/

Seome examples of makefile efficient installations:

1. https://github.com/bxcodec/go-clean-arch/blob/e06c6d0cb37069b0ef56e3df67f80ca130a1ab82/misc/make/tools.Makefile#L85 
2. https://github.com/hyperledger/fabric/blob/111cff51600d26d4b4b05f52825da11e7629e971/gotools.mk#L36
3. https://github.com/tendermint/tendermint/blob/35581cf54ec436b8c37fabb43fdaa3f48339a170/scripts/mockery_generate.sh#L9-L15

Reply:
thank you, at the end I've followed the Docker approach so it works both in local and in the GH action